apps/src/templates/teacherNavigation/PageHeader.tsx

```diff
@@ -28,7 +34,24 @@ const PageHeader: React.FC = () => {
   const isLoadingSectionData = useAppSelector(
     state => state.teacherSections.isLoadingSectionData
   );
+  const [ageGatedModalOpen, setAgeGatedModalOpen] = useState(false);
+  const toggleAgeGatedModal = () => {
+    setAgeGatedModalOpen(!ageGatedModalOpen);
+  };
   const selectedSection = useAppSelector(selectedSectionSelector);
+  const dispatch = useAppDispatch();
+  useEffect(() => {
+    dispatch(loadSectionStudentData(selectedSection.id));
+  }, [dispatch, selectedSection?.id]);
+  const studentData = useAppSelector(
+    state => state.manageStudents?.studentData
+  );
+  const ageGatedStudents = filterAgeGatedStudents(
+    convertStudentDataToArray(studentData)
+  );
+  const ageGatedStudentsUsState =
```

Comment:
This is a lot of work to be doing on the front-end for each teacher-dashboard page just to get a boolean if any student is under 13.

What do you think about creating a new BE endpoint that would return a boolean flag if students in a specified section are under 13?

Reply:
This would also allow the banner to be loaded in async and completely unrelated to all other page content.

Reply:
> This is a lot of work to be doing on the front-end for each teacher-dashboard page just to get a boolean if any student is under 13.

What do you mean by "a lot of work"? Although the code you highlighted in particular is calculating a boolean, the banner also has a modal which shows a list of all the students who are age gated. The banner also shows the date which students are going to be age gated.

> This would also allow the banner to be loaded in async and completely unrelated to all other page content.

Maybe I'm misunderstanding how the code works, but a I think the banner is already loading asynchronously. It doesn't render until the AJAX request for students succeeds.

Reply:
Ah, I didn't realize we needed the students for the banner.